### PR TITLE
MM-37533 - Empty checklists after filtering should omit the body

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Return null if no visible tasks

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-37533

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/rhs/rhs_checklists.tsx [hunk 1]
```diff
@@ -127,7 +127,10 @@ const RHSChecklists = (props: Props) => {
                     collapsed={Boolean(checklistsState[checklistIndex])}
                     setCollapsed={(newState) => dispatch(setChecklistCollapsedState(channelId, checklistIndex, newState))}
                 >
-                    <ChecklistContainer className='checklist'>
+                    <ChecklistContainer
+                        className='checklist'
+                        empty={noVisibleTasks(checklist, checklistItemsFilter, myUser.id)}
```
Comment (cpoile) [inline webapp/src/components/rhs/rhs_checklists.tsx:132]: This, vs returning a null. Open to doing it the other way.

File: webapp/src/components/rhs/rhs_checklists.tsx [hunk 2]
```diff
@@ -127,7 +127,10 @@ const RHSChecklists = (props: Props) => {
                     collapsed={Boolean(checklistsState[checklistIndex])}
                     setCollapsed={(newState) => dispatch(setChecklistCollapsedState(channelId, checklistIndex, newState))}
                 >
-                    <ChecklistContainer className='checklist'>
+                    <ChecklistContainer
```
Comment (lieut-data) [inline webapp/src/components/rhs/rhs_checklists.tsx:130]: Would it make sense to omit rendering this altogether?

**Review by lieut-data — APPROVED** [review]

File: webapp/src/components/rhs/rhs_checklists.tsx [hunk 3]
```diff
@@ -238,9 +241,13 @@ const MainTitle = styled.div`
...
 `;
 
-const ChecklistContainer = styled.div`
+const ChecklistContainer = styled.div<{empty: boolean}>`
     background-color: var(--center-channel-bg);
     padding: 16px 12px;
+
+    ${(props) => props.empty && css`
+        padding: 0px;
+    `}
```
Comment (agarciamontoro) [inline webapp/src/components/rhs/rhs_checklists.tsx:250]: Super nit: I prefer to embed this into the property itself if possible. Something like:

```ts
padding: ${(props.empty) => props.empty ? '0px' : '16px 12px';
```

Or even more succintly:

```ts
padding: ${({empty}) => empty ? '0' : '16px 12px';
```

**Review by agarciamontoro — APPROVED** [review]
Elegant!

File: webapp/src/components/rhs/rhs_checklists.tsx (hunk 3 above)
Comment (cpoile) [reply webapp/src/components/rhs/rhs_checklists.tsx]: Decided to return null, so this is gone. Good advice though, will do in the future. :)

File: webapp/src/components/rhs/rhs_checklists.tsx (hunk 2 above)
Comment (cpoile) [reply webapp/src/components/rhs/rhs_checklists.tsx]: Yah, I was thinking that. Will do that.

**Pull request merged**